hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertRunner.java

```diff
@@ -241,15 +254,96 @@ private static String format(Long time) {
+
+        if (StringUtils.isNotBlank(jobAlert.email)) {
+            sendEmailAlert(jobAlert, alertLink, reason, errors);
+        }
+
+        if (StringUtils.isNotBlank(jobAlert.webhookURL)) {
+            sendWebhookAlert(jobAlert, alertLink, reason, errors);
+        }
+    }
+
+    @VisibleForTesting
+    static Map<Object, Object> getWebhookObject(Spawn spawn,
+                                                AbstractJobAlert jobAlert,
+                                                String alertLink,
+                                                String reason,
+                                                Map<String, String> errors) {
+
+        // Turn all the jobs in error into a list of information about each job
+        List<JSONObject> errorJobInfoList = Lists.newArrayList();
+        errors.forEach((jobUUID, errMsg) -> {
+            JSONObject jobInfo = new JSONObject();
+
+            Job job = spawn.getJob(jobUUID);
+            if (job != null) {
+                try {
+                    jobInfo = Spawn.getJobUpdateEvent(job);
+                    jobInfo.put("state", job.getState())
+                           .put("description", job.getDescription())
+                           .put("start_time", job.getStartTime())
+                           .put("end_time", job.getEndTime());
+                } catch (Exception ex) {
+                    log.error("unable to get job info for webhook alert for job {}", jobUUID, ex);
+                }
+            }
+
+            // Even if we couldn't find the job, we at least know these values to report.
+            try {
+                jobInfo.put("id", jobUUID)
+                       .put("error", errMsg)
+                       .put("cluster", clusterHead);
+            } catch (JSONException ex) {
+                log.error("unexpected error when putting error message into webhook alert", ex);
+            }
+
+            errorJobInfoList.add(jobInfo);
+        });
+
+        return ImmutableMap.builder()
+                           .put("alert_link", alertLink)
+                           .put("alert_type", jobAlert.getTypeString().trim())
+                           .put("alert_reason", reason.trim())
+                           .put("alert_description", jobAlert.description)
+                           .put("jobs_in_error", errorJobInfoList)
+                           .build();
+    }
+
+    private void sendWebhookAlert(AbstractJobAlert jobAlert,
+                                  String alertLink,
+                                  String reason,
+                                  Map<String, String> errors) {
+
+        try {
+            byte[] body = objectMapper.writeValueAsBytes(getWebhookObject(spawn, jobAlert, alertLink, reason, errors));
+            HttpResponse response = HttpUtil.httpPost(jobAlert.webhookURL, ContentType.APPLICATION_JSON.getMimeType(), body, 5_000);
+            if (response.getStatus() / 100 != 2) {
```

Comment:
redirect won't return a 3xx status code when enabled in an apache client. it will return the status code of the request after the redirect.